### PR TITLE
Add new command plf to print the RzIL of the function

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

+ expose API rz_core_analysis_bytes_il
+ add new command  plf
+ add test test/db/cmd/cmd_plf

**Test plan**

CI is green

**Closing issues**

partially address https://github.com/rizinorg/rizin/issues/3669

### CODE REVIEW

File: librz/core/cmd/cmd_analysis.c [hunk 1]
```diff
@@ -5544,7 +5544,7 @@ static void core_colorify_il_statement(RzConsContext *ctx, const char *il_stmt,
 	rz_cons_newline();
 }
 
-static void core_analysis_bytes_il(RzCore *core, const ut8 *buf, int len, int nops, bool pretty) {
+RZ_API void rz_core_analysis_bytes_il(RzCore *core, const ut8 *buf, int len, int nops, bool pretty) {
```
Comment (XVilka) [inline librz/core/cmd/cmd_analysis.c]: Please add Doxygen then

Comment (XVilka) [reply librz/core/cmd/cmd_analysis.c]: And add `rz_return_if_fail(core && buf)`

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -3966,6 +3967,46 @@ RZ_IPI RzCmdStatus rz_cmd_disassembly_function_handler(RzCore *core, int argc, c
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_IPI RzCmdStatus rz_print_function_rzil_handler(RzCore *core, int argc, const char **argv) {
+	ut64 start, end, size, oldoff;
+	RzAnalysisFunction *fcn;
+	RzList *list;
```
Comment (Rot127) [inline librz/core/cmd/cmd_print.c]: You can move those declarations where they are used. Reduces line count and it is easier to see what happens down below.

File: librz/core/cmd/cmd_print.c [hunk 3]
```diff
@@ -3966,6 +3967,46 @@ RZ_IPI RzCmdStatus rz_cmd_disassembly_function_handler(RzCore *core, int argc, c
...
+
+	oldoff = core->offset;
+	list = rz_analysis_get_functions_in(core->analysis, core->offset);
+	if (rz_list_empty(list)) {
+		RZ_LOG_ERROR("No function found in 0x%08" PFMT64x ".\n", core->offset);
+		goto exit;
+	}
+	if (rz_list_length(list) > 1) {
+		RZ_LOG_ERROR("Multiple overlapping functions found at 0x%08" PFMT64x ". "
```
Comment (Rot127) [inline librz/core/cmd/cmd_print.c]:
```suggestion
		RZ_LOG_ERROR("Multiple overlapping functions found at 0x%" PFMT64x ". "
```
Better not limit the address to 8 digits. Address spaces with >32bit are not uncommon.

File: librz/core/cmd/cmd_print.c (hunk 2 above)
Comment (wargio) [reply librz/core/cmd/cmd_print.c:3973]: there is really no reason. but they should be initialized at most to 0.

File: librz/core/cmd/cmd_analysis.c (hunk 1 above)
Comment (wargio) [reply librz/core/cmd/cmd_analysis.c]: add also RZ_NULLABLE/NONNULL etc..

**Comment by PeiweiHu** [comment]
Thanks for the constructive comments!

File: librz/core/cmd/cmd_analysis.c [hunk 4]
```diff
@@ -5544,7 +5544,19 @@ static void core_colorify_il_statement(RzConsContext *ctx, const char *il_stmt,
...
+ *      instructions RzIL of data in \p buf, restricted by
+ *      \p len and \p nops at the same time
+ *
+ * \param core RzCore
+ * \param buf Store the source to be parsed and printed
+ * \param len Maximum length read from \p buf
+ * \param nops Maximum number of instruction, set 0 to disable this
+ * \param pretty Pretty-printing
+ */
+RZ_API void rz_core_analysis_bytes_il(RZ_NONNULL RzCore *core, RZ_NONNULL const ut8 *buf, int len, int nops, bool pretty) {
```
Comment (Rot127) [inline librz/core/cmd/cmd_analysis.c]: - Maybe rename `nops` to `num_ops`. It sounds to similar to the `nop` instruction. And less mental burden is always better :)
- `nops` can be `ut32` if it is not `< 0`.

File: librz/core/cmd/cmd_analysis.c [hunk 5]
```diff
@@ -5544,7 +5544,19 @@ static void core_colorify_il_statement(RzConsContext *ctx, const char *il_stmt,
 	rz_cons_newline();
 }
 
-RZ_API void rz_core_analysis_bytes_il(RzCore *core, const ut8 *buf, int len, int nops, bool pretty) {
+/**
+ * \brief Parse and print \p len bytes and \p nops
+ *      instructions RzIL of data in \p buf, restricted by
```
Comment (Rot127) [inline librz/core/cmd/cmd_analysis.c]: `instruction RzIL` -> `RzIL instructions`?

File: librz/core/cmd/cmd_analysis.c [hunk 6]
```diff
@@ -5544,7 +5544,19 @@ static void core_colorify_il_statement(RzConsContext *ctx, const char *il_stmt,
...
+ *      \p len and \p nops at the same time
+ *
+ * \param core RzCore
+ * \param buf Store the source to be parsed and printed
+ * \param len Maximum length read from \p buf
```
Comment (Rot127) [inline librz/core/cmd/cmd_analysis.c]: `Maximum length read from \p buf in bytes.`

File: librz/core/cmd/cmd_analysis.c [hunk 7]
```diff
@@ -5544,7 +5544,19 @@ static void core_colorify_il_statement(RzConsContext *ctx, const char *il_stmt,
...
 }
 
-RZ_API void rz_core_analysis_bytes_il(RzCore *core, const ut8 *buf, int len, int nops, bool pretty) {
+/**
+ * \brief Parse and print \p len bytes and \p nops
+ *      instructions RzIL of data in \p buf, restricted by
+ *      \p len and \p nops at the same time
+ *
+ * \param core RzCore
+ * \param buf Store the source to be parsed and printed
```
Comment (Rot127) [inline librz/core/cmd/cmd_analysis.c]: Maybe: `\param buf Buffer holding the instruction bytes.`

File: librz/core/cmd/cmd_analysis.c [hunk 8]
```diff
@@ -5544,7 +5544,19 @@ static void core_colorify_il_statement(RzConsContext *ctx, const char *il_stmt,
...
+ * \param len Maximum length read from \p buf
+ * \param nops Maximum number of instruction, set 0 to disable this
```
Comment (Rot127) [inline librz/core/cmd/cmd_analysis.c]: `..., set to 0 to disable it (only use \p len).`

**Review by Rot127 — COMMENTED** [review]
Sorry. Many corrections. Read a lot about code documentation the last days. Now I am super obsessed with it :D

File: librz/core/cmd/cmd_print.c [hunk 9]
```diff
@@ -3966,6 +3967,43 @@ RZ_IPI RzCmdStatus rz_cmd_disassembly_function_handler(RzCore *core, int argc, c
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_IPI RzCmdStatus rz_print_function_rzil_handler(RzCore *core, int argc, const char **argv) {
+
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: Unnecessary space

**Comment by PeiweiHu** [comment]
@Rot127 Thank you for your revision. Adopt them all.

**Review by XVilka — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Review by Rot127 — APPROVED** [review]

**Pull request merged**